src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
```

Comment:
:+1: :+1: 

> because php_phongo_read_preference_prep_tagsets() may end up modifying the zval

do you mean by converting it into object  `convert_to_object(tagSet);` ?